Source/PCH/CMakeLists.txt

```
@@ -15,8 +21,28 @@ target_compile_options(build_pch PRIVATE /Ycpch.h)
 target_compile_options(build_pch PUBLIC /Fp$<TARGET_FILE_DIR:build_pch>/dolphin.pch )
 
 # Sharing a PCH breaks pdb files. So we use the /Z7 option to inline the pdb into
-# the binary. That also requires us to disable minimal rebuilds.
-target_compile_options(build_pch PUBLIC /Z7 /Gm-)
+# the binary. However MSVC gets noisy if you set both /Zi and /Z7
+if (POLICY CMP0141)
+    # CMake 3.25 has a policy that makes us control this somewhat sanely
+    set_property(TARGET build_pch PROPERTY MSVC_DEBUG_INFORMATION_FORMAT "$<$<CONFIG:Debug,RelWithDebInfo>:Embedded>")
+
+    # unfortnually, properties don't properate. So we also do the same globall in the parent
+    set(CMAKE_MSVC_DEBUG_INFORMATION_FORMAT "$<$<CONFIG:Debug,RelWithDebInfo>:Embedded>" PARENT_SCOPE)
+else()
+    if (CMAKE_CXX_FLAGS_DEBUG MATCHES "/Zi")
+        # Otherwise we do an urgy string replace to remove it from FLAGS_DEBUG
```

I don't have anything particularly meaningful to add apart from this, the CMakeSettings change is good (although we should really switch that over to CMakePresets).
```suggestion
        # Otherwise we do an ugly string replace to remove it from FLAGS_DEBUG
```